Mathlib/AlgebraicGeometry/GammaSpecAdjunction.lean

```diff
@@ -0,0 +1,530 @@
+#align algebraic_geometry.LocallyRingedSpace.to_Γ_Spec_c_app_iff AlgebraicGeometry.LocallyRingedSpace.toΓSpecCApp_iff
+
+theorem toΓSpecCApp_spec : toOpen _ (basicOpen r) ≫ X.toΓSpecCApp r = X.toToΓSpecMapBasicOpen r :=
+  (X.toΓSpecCApp_iff r _).2 rfl
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.LocallyRingedSpace.to_Γ_Spec_c_app_spec AlgebraicGeometry.LocallyRingedSpace.toΓSpecCApp_spec
+
+/-- The sheaf hom on all basic opens, commuting with restrictions. -/
+@[simps app]
+def toΓSpecCBasicOpens :
+    (inducedFunctor basicOpen).op ⋙ (structureSheaf (Γ.obj (op X))).1 ⟶
+      (inducedFunctor basicOpen).op ⋙ ((TopCat.Sheaf.pushforward X.toΓSpecBase).obj X.𝒪).1 where
+  app r := X.toΓSpecCApp r.unop
+  naturality r s f := by
+    apply (StructureSheaf.to_basicOpen_epi (Γ.obj (op X)) r.unop).1
+    simp only [← Category.assoc]
+    erw [X.toΓSpecCApp_spec r.unop]
+    convert X.toΓSpecCApp_spec s.unop
+    symm
+    apply X.presheaf.map_comp
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.LocallyRingedSpace.to_Γ_Spec_c_basic_opens AlgebraicGeometry.LocallyRingedSpace.toΓSpecCBasicOpens
+
+/-- The canonical morphism of sheafed spaces from `X` to the spectrum of its global sections. -/
+@[simps]
+def toΓSpecSheafedSpace : X.toSheafedSpace ⟶ Spec.toSheafedSpace.obj (op (Γ.obj (op X))) where
+  base := X.toΓSpecBase
+  c :=
+    TopCat.Sheaf.restrictHomEquivHom (structureSheaf (Γ.obj (op X))).1 _ isBasis_basic_opens
+      X.toΓSpecCBasicOpens
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.LocallyRingedSpace.to_Γ_Spec_SheafedSpace AlgebraicGeometry.LocallyRingedSpace.toΓSpecSheafedSpace
+
+-- Porting Note: Now need much more hand holding: all variables explicit, and need to tidy up
+-- significantly, was `TopCat.Sheaf.extend_hom_app _ _ _ _`
+theorem toΓSpecSheafedSpace_app_eq :
+    X.toΓSpecSheafedSpace.c.app (op (basicOpen r)) = X.toΓSpecCApp r := by
+  have := TopCat.Sheaf.extend_hom_app (Spec.toSheafedSpace.obj (op (Γ.obj (op X)))).presheaf
+    ((TopCat.Sheaf.pushforward X.toΓSpecBase).obj X.𝒪)
+    isBasis_basic_opens X.toΓSpecCBasicOpens r
+  dsimp at this
+  rw [←this]
+  dsimp
+  congr
+
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.LocallyRingedSpace.to_Γ_Spec_SheafedSpace_app_eq AlgebraicGeometry.LocallyRingedSpace.toΓSpecSheafedSpace_app_eq
+
+theorem toΓSpecSheafedSpace_app_spec (r : Γ.obj (op X)) :
+    toOpen (Γ.obj (op X)) (basicOpen r) ≫ X.toΓSpecSheafedSpace.c.app (op (basicOpen r)) =
+      X.toToΓSpecMapBasicOpen r :=
+  (X.toΓSpecSheafedSpace_app_eq r).symm ▸ X.toΓSpecCApp_spec r
+set_option linter.uppercaseLean3 false in
+#align algebraic_geometry.LocallyRingedSpace.to_Γ_Spec_SheafedSpace_app_spec AlgebraicGeometry.LocallyRingedSpace.toΓSpecSheafedSpace_app_spec
+
+-- Porting note : added this helper lemma to help compile `toStalk_stalkMap_to_Γ_Spec`
+theorem toΓSpecSheafedSpace_app_spec_comp {Z} (r : Γ.obj (op X)) (f : _ ⟶ Z) :
```

Comment:
This should be called `_assoc`, not `_comp`, and can probably be autogenerated by putting `@[reassoc]` on the previous theorem.